src/game.cpp

```diff
@@ -5506,6 +5506,19 @@ void game::clear_zombies()
  */
 bool game::spawn_hallucination( const tripoint &p )
 {
+    if( one_in( 100 ) ) {
+        std::shared_ptr<npc> tmp = std::make_shared<npc>();
+        tmp->normalize();
+        tmp->randomize( NC_HALLU );
+        tmp->spawn_at_precise( { get_levx(), get_levy() }, p );
+        if( !critter_at( p, true ) ) {
```

Comment:
I think something like this might work better:
```
    const string_id<npc_template> npc_hallucination( "hallucination" );
    const int halluc_id = g->m.place_npc( get_levx(), get_levy(), true );
    g->load_npcs();

    npc *halluc_npc = g->find_npc( halluc_id );
    if( halluc_npc ) {
        overmap_buffer.insert_npc( tmp ); 
        return true;
    } else {
        return false;
    }
```
I haven't tested it in this application, though.

Reply:
I have a few question : 
- Do I need to add an entry in npc.json to be able to use place_npc ? Basically where does this template is defined ?
- Won't place_npc make the hallucination be static NPCs ?
- What does `overmap_buffer.insert_npc( tmp );` mean ? Because `overmap_buffer.insert_npc` already happens within place_npc

Reply:
Yes, you'd need to define the template in npc.json.

Static NPCs can be companions, move around the map, etc.  The static versus random thing relates to the functions that create them, not their behavior afterwards.

You don't need `insert_npc()` then; I said I hadn't tested it.